### ISSUE TITLE
Stale library versions?

### ISSUE THREAD

**Issue by janstary** [erikd/libsamplerate#32]
Compiling 0.1.9 says

```
/bin/sh ../libtool  --tag=CC   --mode=link /usr/bin/clang  -pipe -Os -arch x86_64 -std=gnu99 -W -Wstrict-prototypes -Wmissing-prototypes  -Waggregate-return -Wcast-align -Wcast-qual -Wnested-externs -Wshadow -Wpointer-arith -pipe -no-undefined -version-info 1:8:1  -L/opt/local/lib -Wl,-headerpad_max_install_names -arch x86_64 -o libsamplerate.la -rpath /opt/local/lib samplerate.lo src_sinc.lo  src_zoh.lo src_linear.lo  -lm -lm
```

What is `-version-info 1:8:1`?

It also says

```
libtool: link: /usr/bin/clang -dynamiclib  -o .libs/libsamplerate.0.dylib  .libs/samplerate.o .libs/src_sinc.o .libs/src_zoh.o .libs/src_linear.o   -L/opt/local/lib -lm  -Os -arch x86_64 -Wl,-headerpad_max_install_names -arch x86_64   -install_name  /opt/local/lib/libsamplerate.0.dylib -compatibility_version 2 -current_version 2.8 -Wl,-single_module^M^M
```

What is `-compatibility_version 2`?

**Comment by erikd** [comment]
No idea. On Linux the link line looks like:
```
/bin/bash ./libtool  --tag=CC   --mode=link gcc  -O2 -pipe -Werror -W \
    -Wmissing-prototypes -Wall -Waggregate-return -Wcast-align -Wcast-qual \
    -Wnested-externs -Wshadow -Wpointer-arith -no-undefined -version-info \
    1:9:0 -Wl,--version-script=./src/Version_script -Wl,-O1 -Wl,--as-needed \
    -Wl,--no-undefined -Wl,--gc-sections -o src/libsamplerate.la \
    -rpath /usr/local/lib src/samplerate.lo src/src_sinc.lo src/src_zoh.lo \
    src/src_linear.lo  -lm 

libtool: link: gcc -shared  -fPIC -DPIC  src/.libs/samplerate.o src/.libs/src_sinc.o \
    src/.libs/src_zoh.o src/.libs/src_linear.o   -lm  -O2 \
    -Wl,--version-script=./src/Version_script \
    -Wl,-O1 -Wl,--as-needed -Wl,--no-undefined -Wl,--gc-sections   -Wl,-soname \
   -Wl,libsamplerate.so.1 -o src/.libs/libsamplerate.so.1.0.9
```

**Comment by janstary** [comment]
Are you compiling 0.1.9 or git?

**Comment by erikd** [comment]
From Git.

**Comment by janstary** [comment]
So it's not a difference between Linux and elsewhere, but 0.1.9 and git.

The 1:8:1 only appears in 0.1.9, in `./configure.ac:SHARED_VERSION_INFO="1:8:1"`.
In git, it's `./configure.ac:SHARED_VERSION_INFO="1:$VERSION_MINOR:0"`

**Comment by erikd** [comment]
It should probably be:
```
SHARED_VERSION_INFO="1:$VERSION_MINOR:1"
```
to indicate that an interface has been added.

**Comment by erikd** [comment]
Fixed in 02ebb9f0fc4d67c . Thanks!
